pkg/asset/installconfig/clusterid.go

```diff
@@ -74,6 +75,15 @@ func generateInfraID(base string, maxLen int) string {
 	}
 	base = strings.TrimRight(base, "-")
 
+	// Return the InfraID provided by the "INFRA_ID" environment variable if valid
+	if envInfraID, ok := os.LookupEnv("INFRA_ID"); ok && envInfraID != "" {
+		validInfraIDString := fmt.Sprintf("^%s-([b-df-hj-np-tv-z0-9]){%d}$", base, randomLen)
+		validInfraIDRegex := regexp.MustCompile(validInfraIDString)
+		if validInfraIDRegex.MatchString(envInfraID) {
+			return envInfraID
```

Comment:
Incase there is no match you need to return with an error.
Think that the user set the "OPENSHIFT_INSTALL_INFRA_ID_OVERRIDE" env var, he doesn't need to wait till the installation starts to see that the env var is not good

Reply:
Ok